### PR TITLE
US112279: Outcomes Picker: Search bar + bold text

### PR DESCRIPTION
**Pull request by lvyuan95-d2l**

### CODE REVIEW

File: d2l-select-outcomes-hierarchical-list.js [hunk 1]
```diff
@@ -93,5 +131,82 @@ Polymer({
...
 
 		return hierarchyRoot;
-	}
+	},
+
+	_getDisplayedHierarchyItems: function(entity, searchText) {
+		if (!entity) return [];
+		if (searchText === undefined) return entity;
+		if (searchText === '') {
+			IronA11yAnnouncer.instance.fire('iron-announce',
```
Comment (rgoessl) [inline d2l-select-outcomes-hierarchical-list.js]: Did you test this? I couldn't get `IronA11yAnnouncer.instance` to work in my case.

File: d2l-select-outcomes-hierarchical-list.js [hunk 2]
```diff
@@ -93,5 +131,82 @@ Polymer({
...
+			'search-results-changed',
+			{ bubbles: true, composed: false, detail: { value: numOfLeaves } }
+		));
+
+		return filtered;
+	},
+
+	_search: function(entity, searchText = '') {
+		const description = (entity && entity.properties && entity.properties.description)
+			? entity.properties.description.toLowerCase().normalize()
```
Comment (rgoessl) [inline d2l-select-outcomes-hierarchical-list.js]: nit: We have a behaviour to get these props `import OutcomeParserBehaviour from 'd2l-activity-alignments/d2l-outcome-parser-behavior.js';`

File: d2l-select-outcomes-hierarchical-list.js [hunk 3]
```diff
@@ -93,5 +131,82 @@ Polymer({
...
+			entity.entities = filteredSublevels;
+			return filteredSublevels.length !== 0 ? entity : null;
+		}
+	},
+
+	_applyBoldText: function(entity, searchText) {
+		if (!entity || !searchText) return entity;
+
+		for (const i of searchText.split(' ').filter(i => i)) {
+			const searchRegex = new RegExp(i, 'ig');
```
Comment (rgoessl) [inline d2l-select-outcomes-hierarchical-list.js]: Should escape `i`
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#Escaping

File: d2l-outcome-hierarchy-item.js [hunk 4]
```diff
@@ -411,6 +415,10 @@ Polymer({
 	_setIsSelectedState: function(item, alignments) {
```
[the remark this answers is not in the document]
Comment (lvyuan95-d2l) [reply d2l-outcome-hierarchy-item.js]: For the current implementation to set the selected state of cells, when searchText changed, the result will change too. But it doesn't trigger the cell to rerender its selected state

File: d2l-select-outcomes-hierarchical-list.js (hunk 1 above)
Comment (lvyuan95-d2l) [reply d2l-select-outcomes-hierarchical-list.js]: Yeah it works

File: d2l-select-outcomes-hierarchical-list.js (hunk 3 above)
Comment (lvyuan95-d2l) [reply d2l-select-outcomes-hierarchical-list.js]: You mean something like `new RegExp(escapeRegExp(i), 'ig');` ?

Comment (rgoessl) [reply d2l-select-outcomes-hierarchical-list.js]: yes

**Pull request merged**